### PR TITLE
[ENH] extend data loaders to handle regression `ts` files, add Tecator dataset for Time Series Regression

### PR DESCRIPTION
**Pull request by JonathanBechtel**
#### Reference Issues/PRs
#4314

#### What does this implement/fix? Explain your changes.
I added a new dataset that can be used for Time Series Regression, since they were lacking in the current set of example datasets.  This will allow for better documentation and examples for various algorithms that make use of it.

#### Does your contribution introduce a new dependency? If yes, which one?
No new dependencies.

#### What should a reviewer concentrate their feedback on?
I had to make some modifications to the standard metadata used in `.ts` files, since the data loading functions are all built with the assumption that it's a classification dataset.  The function `load_from_tsfile_to_dataframe` has a lot of conditional logic, much of which is based off the assumption that the target column is populated with discreet labels.  So I added the flag `@problemType`, which can be either `regression` or `classification`, if the flag is set to `@regression`, then you can skip to some additional conditional logic that transforms the target column to a numeric output.  

It might be notable that I did NOT add this flag to every `.ts` file, and just allowed the logic of the ts file loading function to assume that everything is a classification dataset unless otherwise noted.

#### Did you add any tests for the change?
I added the function `test_load_tecator`, which tests for all of the unique values that can be passed in for the arguments `split`, `return_X_y`, and `return_type`.  This hasn't been done for the other data loading functions, but on first principles it strikes me as a good idea.

#### Any other comments?
If this gets accepted I'll go ahead and update some other documentation that could make use of this.

### CODE REVIEW

File: sktime/datasets/tests/test_data_io.py [hunk 1]
```diff
@@ -124,6 +125,45 @@ def test_load_basic_motions(return_X_y, return_type):
     assert valid, msg
 
 
+@pytest.mark.parametrize("return_X_y", [True, False])
+@pytest.mark.parametrize(
+    "return_type", [mtype for mtype in MTYPE_LIST_PANEL if mtype not in _TO_DISABLE]
+)
+@pytest.mark.parametrize("split", ["TRAIN", "TEST", None])
+def test_load_tecator(return_X_y, return_type, split):
+    """Test load_basic_motions function to check for proper loading.
```
Comment (fkiraly) [inline sktime/datasets/tests/test_data_io.py:134]: typo

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks, useful!

* could you kindly add a reference to the dataset loader, where it's from, papers to cite, etc?
* looks like the `ts` file loader is changed - does this change the specification? It looks like this change is downwards compatible as it adds a "problemtype" identifier, but it would be better if we do not change the file specification. Is this needed?
* if we want to change the specification, we should change the specification document and increment the version, in `ts.rst`.

**Comment by JonathanBechtel** [comment]
@fkiraly Thanks.  

Regarding the dataset loader -- I assume you mean the academic information surrounding the Tecator dataset itself?  References for that are added in the dataset description in the `.ts` files, but I can add them to `load_tecator` if that's what you were referring to.

Regarding. the `.ts` file loader -- it was the simplest way to incorporate the use of a regression dataset with a `.ts` file format.  The problem came with the `classLabel` flag.  If you set it to false, it returned results that were incoherent -- `X` and `y` were returned as strings instead of the intended data type.  If you set it to `True`, it wants the specific class labels to be mentioned in the specification, which doesn't really make sense for this type of dataset.  

The other issue is that `y` values are returned as strings as a default, which also doesn't make sense for this dataset.  

So I think there's a certain amount of configuration you have to do regardless.  One idea, if you did not want to add new metadata, is for the `classLabel` flag, if the value it set to `numeric`, you could just use that to set the `is_regression_problem` variable to True inside the data loading function.  That should work and would remove the need to have a new specification for a dataset.

**Comment by fkiraly** [comment]
> but I can add them to load_tecator if that's what you were referring to.

Yes, that's what I meant - the contents of the `ts` file are typically not visible to the user.

> was the simplest way to incorporate the use of a regression dataset

I see! We should then extend the specification to regression datasets.

**Comment by fkiraly** [comment]
PS: I think your changes conflict with @hazrulakmal's recent refactor of the data-IO module. Does not look like major work to resolve, but still GitHub needs to be made happy...

**Comment by JonathanBechtel** [comment]
@fkiraly The conflict is due to a print statement that I removed:

```
<<<<<<< regression-dataset
    with open(full_file_path_and_name, "r", encoding="utf-8") as f:
=======
    # print(full_file_path_and_name)
    with open(full_file_path_and_name, encoding="utf-8") as f:
>>>>>>> main
```

I do believe I can simplify what I had done previously, and can remove the `@problemType` flag, but it's not possible (far as I can tell) to completely re-use some of the `.ts` data loading functions for regression datasets, since it conflicts with their current assumptions.

**Comment by hazrulakmal** [comment]
>  I think your changes conflict with @hazrulakmal's recent refactor of the data-IO module. 

are you referring to this #5239? I don't think that's the reason for the conflicts as the PR is not merged to the main yet.

**Comment by JonathanBechtel** [comment]
> However, if you are not referring to the conflicts in this PR but rather a general conflict then yes there is a conflict if one of these two PRs gets merged later on. One of us, either @JonathanBechtel or me, has to update the PR once one of these two PRs is merged. so I invite Jonathan to look at the cleanup PR that I raised in case you have anything to comment on in regard to the module file restructuring.

@hazrulakmal I don't see how my current PR would conflict with that right now since they're both open.  As fas as I can tell the current conflict is only due to the issue that I highlighted above (according to Github).  

@fkiraly If there are plans to merge the one by @hazrulakmal I agree I should wait for that one and then add mine to make sure they're harmonious.  His change seems more substantive than mine, so it seems like his ought to have priority.

**Comment by fkiraly** [comment]
the conflict seems quite simple, someone must have removed the commented out print statement, possibly due to linter complaint, next to a line that was changed. We should be safe if we force this PR's version through, possibly handling the print comment for the linter's sake.

**Comment by JonathanBechtel** [comment]
> the conflict seems quite simple, someone must have removed the commented out print statement, possibly due to linter complaint, next to a line that was changed. We should be safe if we force this PR's version through, possibly handling the print comment for the linter's sake.

I removed the `print` statement, since I figured a commented out `print` statement isn't supposed to be there.  It's easy enough to add it back in, but I imagine it's there inadvertently.

**Comment by achieveordie** [comment]
@JonathanBechtel I added `@targetlabel` a while back to accommodate regression datasets [here](https://github.com/sktime/sktime/pull/3436), maybe we should think a bit more about updating the specifications to include regression datasets as well since we don't currently document it well. We'd probably also have to update our code logic as you rightfully mentioned.

There's also an adjacent issue I'd like to bring to everyone's notice: #3521

**Comment by fkiraly** [comment]
There were some conflicts with @hazrulakmal's refactor, I spliced this PR into those 1:1, hope this all worked out...

**Comment by fkiraly** [comment]
@JonathanBechtel, this tries to find the tecator dataset at the UEA TSC repository. Is that correct? Should it not be looking in the sktime repository instead?

I checked the pre-merge state of affairs, and there it was the same afaik.

**Comment by JonathanBechtel** [comment]
@fkiraly It's not pulling from UCR.  There's are train and test files in the `data/datasets/Tecator` folder.  Am coming back to this, will try and look at it today and over the weekend.

**Comment by fkiraly** [comment]
> It's not pulling from UCR

I think it was, it was using `_load_dataset` which does, `_load_provided_dataset` is the utility that looks in the `sktime` local folders. I might be mistaken though, this is not based on deeper knowledge of the module, and @hazrulakmal has recently refactored it.

**Comment by JonathanBechtel** [comment]
@fkiraly It looks like `_load_dataset` checks to see if the file exists locally, and if it does, calls `_load_provided_dataset`, and if not, downloads from the Time Series classification repository.  I suppose it's more succinct just to call `_load_provided_dataset`, but at the time I wrote it I was mostly just sticking to the convention used for `.ts` files.

**Comment by fkiraly** [comment]
> FWIW, most of the .ts files in the datasets module are loaded the same way.

No worries, please change if you prefer.
What I did is I tried to merge this PR with the refactor of @hazrulakmal that moved a lot of code around, including the issues in here.

**Pull request closed without merging**